src/index.ts

```diff
@@ -21,8 +34,40 @@ interface V3Params {
   p: number
 }
 
-function mergeToV3ParamsWithDefaults(params?: Partial<V3Params>): V3Params {
-  const v3Defaults: V3Params = {
+function validateHexString(paramName: string, str: string, length?: number) {
+  if (!str && !length) {
+    return str
+  }
+  if (str.toLowerCase().startsWith('0x')) {
+    str = str.slice(2)
+  }
+  if (!/^[0-9a-f]{2}([0-9a-f]{2})*$/i.test(str)) {
```

Comment:
This is implicitly testing for length >= 2.

Reply:
Right: `ff` or `0xff` would work but not `f` or `0xf`. But I think I'm missing the point you have in mind, can you explain further?

Reply:
Just that if `validateHexString` is called without a length argument (e.g. for `salt`), inputs with length < 2 will still be rejected. I don't expect anyone to provide a `salt` with length < 2, so it should be fine.